libs/sdk-backend-tiger/src/convertors/fromBackend/afm/GrandTotalsConverter.ts

```diff
@@ -0,0 +1,203 @@
+): IResultHeader[][][] {
+    const combinedHeaderItems: IResultHeader[][][] = [];
+    let grandTotalIdx = 0;
+    dataHeaderItems.forEach((dimensionHeader, dimIdx) => {
+        if (totalDimensions.includes(dimIdx)) {
+            combinedHeaderItems.push(dimensionHeader);
+        } else {
+            combinedHeaderItems.push(grandTotalHeaderItems[grandTotalIdx]);
+            grandTotalIdx += 1;
+        }
+    });
+    return combinedHeaderItems;
+}
+
+type ResultData = DataValue[] | ResultData[];
+
+/**
+ * SDK expects grand total data in an array with the lengths [dimensions][totals][values],
+ * but Tiger API returns an array of "Grand Totals", where each may consist of multiple
+ * totals from the dimension definition.
+ * Each grand total has data in the same format as the main result, though the dimensionality will be different.
+ * SDK supports only one-dimensional totals, so we will skip multidimensional totals to simplify the implementation.
+ * The data in the grand total is in an array with the lengths [total dimensions][function][metric*values].
+ * We can determine the original total by a combination of function and metric on a specific dimension.
+ */
+function transformGrandTotal(
+    grandTotal: ExecutionResultGrandTotal,
+    headerItems: Readonly<IResultHeader[][][]>,
+    totalDimensions: number[],
+    definition: IExecutionDefinition,
+): GrandTotal[] {
+    if (totalDimensions.length !== 1) {
+        // If the grand total belongs to multiple or zero dimensions it is either
+        // a grand total from a >2D result or it is a total of totals,
+        // neither of which can currently be handled
+        return [];
+    }
+    // Here we assume that there are only two dimensions in the definition, as we need to find on which dimension
+    // the totals in this grand total result were defined. If there were more than two dimensions, then a total
+    // definition on a single dimension would result in >1 dimensions in `totalDimensions`, which we checked already.
+    // TODO revisit when multidimensional totals are supported
+    const dimensionIdx = definition.dimensions.findIndex((_, idx) => idx !== totalDimensions[0]);
+    const totalDefinitions = definition.dimensions[dimensionIdx]?.totals;
+    if (totalDefinitions === undefined) return [];
+
+    const transformedTotals: Record<number /* total index */, DataValue[]> = {};
+    const resultData = grandTotal.data as ResultData;
+    walk(resultData);
+    return Object.keys(transformedTotals)
+        .map(Number)
+        .map((totalIdx): GrandTotal => {
+            const data = transformedTotals[totalIdx];
+            return {
+                dimensionIdx,
+                totalIdx,
+                data,
+            };
+        });
+
+    /**
+     * Iteratively walk through the data, get headers at each point and use them to assign data values to a total.
+     */
+    function walk(data: ResultData, coordinates: number[] = []) {
```

Comment:
Although this will work thanks to hoisting, please define this function before calling it to make it less confusing.